parser/src/sourcemanager.cpp

```diff
@@ -221,6 +234,31 @@ void SourceManager::updateFile(const model::File& file_)
     });
 }
 
+void SourceManager::removeFile(const model::File& file_)
+{
+  bool removeContent = false;
+
+  // Delete File and FileContent (only when no other File references it)
+  _transaction([&, this]() {
+    auto relFiles = _db->query<model::File>(
+        odb::query<model::File>::content == file_.content.object_id());
+    if(std::distance(relFiles.begin(), relFiles.end()) == 1)
+    {
+      removeContent = true;
+      _db->erase<model::FileContent>(file_.content.object_id());
+    }
+    _db->erase<model::File>(file_.id);
+  });
+
+  // Maintain cache
+  _createFileMutex.lock();
```

Comment:
Shouldn't it be better to use the lock guard (scoped guard) initialized here, in case an exception or something happens, rather than to manual lock?

Reply:
Theoretically you are correct (and refactored in 4bfa0e3011222b14b1e78e95eee12c253c7edbf8), but I would be surprised to have an exception thrown there and even it is thrown, the `SourceManager`'s cache remains in an inconsistent state, so termination of the CodeCompass parser is advised.